### DISCUSSION TITLE
State of all shapes in the layer

### DISCUSSION THREAD

**Discussion by cubecleveland** [geoman-io/leaflet-geoman#1081] [Questions & Help] [answered]
Hi, we are working of a react implementation fo the library after playing with ti for some time in our app.

At the moment we are trying to keep a state of all the shapes in the layer including their coordinates.
Can i get global event where every single layer chnages its position.

I was trying to play with the layer.pm.dragend event but that reuiqres us to attach the handler to every singles layer that is currently their INCLUDING ONES that will be created in the future... not usre if that is the way to go.

any help appreciated.

**Comment by Falke-Design** [comment, marked as answer]
You can add to the Shapes directly a event forwarding. So you don't need to add the listener for each layer.

```
L.Polyline.include({
        _fireOnMap: function(e){
            e.firedLayer = this;
            this._map.fire(e.type,e)
        },
        addLayer: function(){
          if(this._map){
            this.on('pm:dragstart',this._fireOnMap,this)
            this.on('pm:drag',this._fireOnMap,this)
            this.on('pm:dragend',this._fireOnMap,this)
          }
        }
      });
L.Polyline.addInitHook(function(){
        this.on('add',this.addLayer, this);
      });
```

**Reply by cubecleveland** [reply]
wonderful.! 
Are u guys interested in the React-Project as an open source?

**Reply by Falke-Design** [reply]
We have currently not that much time to feature a other project
